### ISSUE TITLE
build fails for cpython 3.11.0a2

### ISSUE THREAD

**Issue by davidism** [python-greenlet/greenlet#273]
It seems something internal has changed between a1 and a2, building greenlet from sdist fails and there is no wheel available yet. The error is `‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘frame’`. https://github.com/pallets/werkzeug/runs/4131743156?check_suite_focus=true#step:9:240

```
❯ pip install --no-cache-dir greenlet

...

  running build_ext
  building 'greenlet._greenlet' extension
  creating build/temp.linux-x86_64-3.11
  creating build/temp.linux-x86_64-3.11/src
  creating build/temp.linux-x86_64-3.11/src/greenlet
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/home/david/.local/share/virtualenvs/67d0b5b3ea2275a/include -I/home/david/.local/python/include/python3.11 -c src/greenlet/greenlet.c -o build/temp.linux-x86_64-3.11/src/greenlet/greenlet.o
  src/greenlet/greenlet.c: In function ‘g_switchstack’:
  src/greenlet/greenlet.c:529:38: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘frame’; did you mean ‘cframe’?
    529 |         current->top_frame = tstate->frame;
        |                                      ^~~~~
        |                                      cframe
  src/greenlet/greenlet.c:578:17: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘frame’; did you mean ‘cframe’?
    578 |         tstate->frame = target->top_frame;
        |                 ^~~~~
        |                 cframe
  error: command '/usr/bin/gcc' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for greenlet
  Running setup.py clean for greenlet
Failed to build greenlet
```

**Comment by jamadden** [comment]
This is fixed in the current trunk that will become 2.0.

In general, released versions of greenlet may or may not work with Python versions that are newer than the release. It often (but not always) requires a new version of greenlet to support a new version of Python.

**Comment by davidism** [comment]
No problem, guess it was just lucky that it happened to build on a1 yesterday. I had our CI ignore that for now.
